From: Shahed Shaikh <shahed.shaikh@qlogic.com>
Date: Mon, 24 Feb 2014 08:54:47 -0500
Subject: [PATCH 211/270] qlcnic: Fix usage of use_msi and use_msi_x module
 parameters
Patch-mainline: v3.14-rc6
Git-commit: b7520d2b59b09eb284e5fc9080d13145f0a8d9fd
References: bsc#909350 FATE#317546

Once interrupts are enabled, instead of using module parameters,
use flags (QLCNIC_MSI_ENABLED and QLCNIC_MSIX_ENABLED) set by driver
to check interrupt mode.

Signed-off-by: Shahed Shaikh <shahed.shaikh@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Gary Ching-Pang Lin <glin@suse.com>
---
 drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
@@ -3857,7 +3857,7 @@ int qlcnic_validate_rings(struct qlcnic_
 		strcpy(buf, "Tx");
 	}
 
-	if (!qlcnic_use_msi_x && !qlcnic_use_msi) {
+	if (!QLCNIC_IS_MSI_FAMILY(adapter)) {
 		netdev_err(netdev, "No RSS/TSS support in INT-x mode\n");
 		return -EINVAL;
 	}
